### DISCUSSION TITLE
Turborepo including PHP backend app?

### DISCUSSION THREAD

**Discussion by andrewfairlie** [vercel/turborepo#1900] [Help] [answered]
Does Turborepo support running PHP apps from the /apps/ folder?

I appreciate that Turborepo's features are JS oriented, but would anything break if we put a backend app or CMS in there?

Disclaimer: I'm just evaluating Turborepo at this point and haven't actively used it.

**Comment by mehulkar** [comment, marked as answer]
There is no restriction on the type of code you put in a monorepo that runs Turborepo. However, if you want Turborepo to _do_ anything in that PHP codebase, you need two things:

- You'll need your monorepo's _package manager_ to recognize the workspace
- Your PHP "workspace" will need to include a package.json that has the `scripts` key with the tasks you want to run. 

For example, if your repo looks like this:


```
apps/
|-- php-app
    |-- package.json
|-- nextjs-app
    |-- package.json
package.json
turbo.json
```

If you're generally using `npm`, the root package.json will declare workspace:

```jsonc
{
  // <rest of package.json omitted>
  "workspaces": [ "apps/*" ]
}
```

and `apps/php-app/package.json` will look like

```jsonc
{
  "name": "php-app",
  "scripts": {
    "build": "composer start" // I haven't worked too much with PHP , so this is just a guess
  }
}
```

and `turbo.json`, could then have a standard `build` task (as shown in docs). 

Hope that helps!

**Reply by andrewfairlie** [reply]
Thank you 🙇‍♂️
